### PR TITLE
Issue 314 Refactor logs command 

### LINKED ISSUE #314: Improve error handling for logs
_From @trisberg on February 14, 2018 16:8_

Running `riff logs` depends on the transient nature of the pod running the function - we should have better error messages when the pod isn't in a state where the logs are accessible, right now you get stuff like:
```
$ riff logs -n square
Displaying logs for container sidecar of function square in namespace default

exit status 1: error: error executing jsonpath "{.items[0].metadata.name}": array index out of bounds: index 0, length 0

Error exit status 1 - Function square may not be currently active
```

would be nice to eliminate the jsonpath error

_Copied from original issue: projectriff/riff-cli#33_

### PR DESCRIPTION
**Pull request by dturanski**
Fixes #314

### CODE REVIEW

**Comment by dturanski** [comment]
Sorry forgot to squash

File: riff-cli/pkg/osutils/osutils.go [hunk 1]
```diff
@@ -122,7 +134,7 @@ func Exec(cmdName string, cmdArgs []string, timeout time.Duration) ([]byte, erro
 	// This time we can simply use Output() to get the result.
 	out, err := cmd.Output()
 	if err != nil {
-		ioutils.Error(fmt.Sprint(err) + ": " + stderr.String())
```
Comment (trisberg) [inline riff-cli/pkg/osutils/osutils.go]: looks like we are loosing the actual error message now.

Used to be:

```
$ ./riff-cli/riff list
Listing function resources in namespace default

exit status 1: the server doesn't have a resource type "functions"

Error: exit status 1
```

Now, the middle error line is missing

**Comment by dturanski** [comment]
I took out a line in osutils that prints the error message. I’ll take another look

**Comment by trisberg** [comment]
I don't see much change for the logs message either other than suppressing the json parsing error. I'd prefer to hold off on this change until Eric is done with the refactoring.

**Comment by trisberg** [comment]
Will address this at some later point.

**Pull request closed without merging**